README.md

```diff
@@ -4,13 +4,129 @@
 [![IRC Channel](https://img.shields.io/badge/irc-%23calico-blue.svg)](https://kiwiirc.com/client/irc.freenode.net/#calico)
 # Project Calico
 
+<!--
 <blockquote>
 Note that the documentation in this repo is targeted at Calico contributors.
 <h1>Documentation for Calico users is here:<br><a href="http://docs.projectcalico.org">http://docs.projectcalico.org</a></h1>
 </blockquote>
+-->
+
+This repository contains the source code for Project Calico's optional Typha daemon, which is
+currently in beta.  An instance of Typha sits between the datastore (such as the
+Kubernetes API server) and many instances of Felix.
+
+![A small cluster of Typha nodes fan out updates to many Felix instances](docs/fan-out.png "A small cluster of Typha nodes fan out updates to many Felix instances  ")
+
+This has many advantages:
+
+- Since one Typha instance can support hundreds of Felix instances, it reduces the load on the datastore
+  by a large factor.
+
+- Since Typha can filter out updates that are not relevant to Felix, it also reduces Felix's
+  CPU usage.  In a high-scale (100+ node) Kubernetes cluster, this is essential because the
+  number of updates generated by the API server scales with the number of nodes.
+
+## How can I try Typha?
+
+We're still in the process of adding Typha to our documentation.  In the meantime, if you're using the
+Kubernetes Datastore Driver (KDD) with Kubernetes 1.6+ and you'd like to try it out, follow the instructions below.
+
+Since Typha has the most impact when using the KDD, we're focusing on that to begin with.  Install the Kubernetes
+specs below to creates a 3-node deployment of Typha and expose them as a service called `calico-typha`.  A three-node
+deployment is enough for ~600 Felix instances.  Typha scales horizontally so feel free to reduce/increase
+the number of replicas.  If you're running a small cluster, you may wish to reduce the CPU reservation
+proportionately.
+
+
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: calico-typha
+      namespace: kube-system
+      labels:
+        k8s-app: calico-typha
+    spec:
+      ports:
+        - port: 5473
+          protocol: TCP
+          targetPort: calico-typha
+          name: calico-typha
+      selector:
+        k8s-app: calico-typha
+    ---
+    apiVersion: extensions/v1beta1
+    kind: Deployment
+    metadata:
+      name: calico-typha
+      namespace: kube-system
+      labels:
+        k8s-app: calico-typha
+    spec:
```

Comment:
You'll need a service account here to give Typha the creds to connect to the API server.

E.g. https://github.com/tigera/kubeadm-deployer/blob/master/calico-kdd-typha.template.yaml#L53

Detail might depend on the wider deployment though.